src/plot/crosshair.js

```diff
@@ -132,14 +133,19 @@ class Crosshair extends PureRenderComponent {
       marginLeft,
       innerWidth,
       innerHeight} = this.props;
+    let {orientation} = this.props;
     const value = getFirstNonEmptyValue(values);
     if (!value) {
       return null;
     }
     const x = getAttributeFunctor(this.props, 'x');
     const innerLeft = x(value);
 
-    const orientation = (innerLeft > innerWidth / 2) ? 'left' : 'right';
+    // infer orientation from x value if not overridden
+    if (!orientation) {
+      orientation = (innerLeft > innerWidth / 2) ? 'left' : 'right';
```

Comment:
Biggest problem with that is that I'll have to do lines 140 and 141 in `getOrientation()`.